packages/eslint-plugin/src/rules/no-useless-default-assignment.ts

```diff
@@ -0,0 +1,263 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+import * as ts from 'typescript';
+
+import {
+  createRule,
+  getParserServices,
+  isTypeAnyType,
+  isTypeFlagSet,
+  isTypeUnknownType,
+} from '../util';
+
+type Options = [
+  {
+    allowRuleToRunWithoutStrictNullChecksIKnowWhatIAmDoing?: boolean;
+  },
+];
+
+type MessageId =
+  | 'noStrictNullCheck'
+  | 'suggestRemoveDefault'
+  | 'uselessDefaultAssignment';
+
+export default createRule<Options, MessageId>({
+  name: 'no-useless-default-assignment',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow default values that will never be used',
+      requiresTypeChecking: true,
```

Comment:
[Feature] What do you think about adding this rule to the `strict` preset? I think it's great and I can't think of any reasons not to enable it besides "our types are inaccurate".

The only reason I'm not suggesting `recommended` is because of the semver breaking change. But for the next major... 🚀